packages/regl-worldview/src/commands/Arrows.js

```diff
@@ -7,87 +7,104 @@
+      quat.rotateY(orientation, orientation, Math.PI / 2);
+      dir = vec3.transformQuat([0, 0, 0], UNIT_X_VECTOR, orientation);
 
-        shaftWidthX = marker.scale.y || 1;
-        shaftWidthY = marker.scale.z || 1;
-        headWidthX = 2 * shaftWidthX;
-        headWidthY = 2 * shaftWidthY;
+      shaftWidthX = marker.scale.y || 1;
+      shaftWidthY = marker.scale.z || 1;
+      headWidthX = 2 * shaftWidthX;
+      headWidthY = 2 * shaftWidthY;
 
-        // these magic numbers taken from
-        // https://github.com/ros-visualization/rviz/blob/57325fa075893de70f234f4676cdd08b411858ff/src/rviz/default_plugin/markers/arrow_marker.cpp#L113
-        headLength = 0.23 * (marker.scale.x || 1);
-        shaftLength = 0.77 * (marker.scale.x || 1);
-      }
+      // these magic numbers taken from
+      // https://github.com/ros-visualization/rviz/blob/57325fa075893de70f234f4676cdd08b411858ff/src/rviz/default_plugin/markers/arrow_marker.cpp#L113
+      headLength = 0.23 * (marker.scale.x || 1);
+      shaftLength = 0.77 * (marker.scale.x || 1);
+    }
 
-      const shaftPosition = vec3.scaleAndAdd([0, 0, 0], basePosition, dir, shaftLength / 2);
-      const headPosition = vec3.scaleAndAdd([0, 0, 0], basePosition, dir, shaftLength + headLength / 2);
+    const shaftPosition = vec3.scaleAndAdd([0, 0, 0], basePosition, dir, shaftLength / 2);
+    const headPosition = vec3.scaleAndAdd([0, 0, 0], basePosition, dir, shaftLength + headLength / 2);
 
-      cylinders.push({
-        scale: { x: shaftWidthX, y: shaftWidthY, z: shaftLength },
-        color: marker.color,
-        pose: {
-          position: vec3ToPoint(shaftPosition),
-          orientation: vec4ToOrientation(orientation),
-        },
-      });
-      cones.push({
-        scale: { x: headWidthX, y: headWidthY, z: headLength },
-        color: marker.color,
-        pose: {
-          position: vec3ToPoint(headPosition),
-          orientation: vec4ToOrientation(orientation),
-        },
-      });
-    }
-    return (
-      <React.Fragment>
-        <Cylinders>{cylinders}</Cylinders>
-        <Cones>{cones}</Cones>
-      </React.Fragment>
-    );
+    cylinders.push({
+      // Set the original marker so we can use it in mouse events
+      originalMarker: marker,
+      scale: { x: shaftWidthX, y: shaftWidthY, z: shaftLength },
+      color: marker.color,
+      pose: {
+        position: vec3ToPoint(shaftPosition),
+        orientation: vec4ToOrientation(orientation),
+      },
+    });
+    cones.push({
+      // Set the original marker so we can use it in mouse events
+      originalMarker: marker,
+      scale: { x: headWidthX, y: headWidthY, z: headLength },
+      color: marker.color,
+      pose: {
+        position: vec3ToPoint(headPosition),
+        orientation: vec4ToOrientation(orientation),
+      },
+    });
   }
+
+  return (
+    <Fragment>
+      <Cylinders getChildrenForHitmap={getChildrenForHitmapWithOriginalMarker} {...passedProps}>
```

Comment:
Thinking about alternative APIs for this. Some ideas:
- `getChildrenForHitmap={makeGetChildrenForHitmap(originalMarker)}` (don't have to mutate the marker itself, although I guess the child needs to pass it down somehow — could do this by keeping a map from draw call to original marker)
- what if we override the original marker with a context, such as `return <Worldview.OriginalMarkerProvider value={originalMarker}> <Cylinders> {... as normal ... } </Cylinders> ... </Worldview.OriginalMarkerProvider>`